fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -285,11 +286,48 @@
         return null;
     });
 
-    private static final List<Function<Quad, IllegalArgumentException>> quadValidators =
-            ImmutableList.<Function<Quad, IllegalArgumentException>>builder()
+
+    private static final Function<Triple, IllegalArgumentException> validateNoMementRdfTypes = uncheck(x ->  {
+        final org.apache.jena.graph.Node object = x.getObject();
+        if (object.isURI() && x.getPredicate().getURI().equals(RDF_TYPE_URI) &&
+            object.getURI().startsWith(MEMENTO_NAMESPACE)) {
+            return new IllegalArgumentException(
+                "The " + RDF_TYPE_URI + " predicate may not take an object in the memento namespace (" +
+                MEMENTO_NAMESPACE + ").");
+        }
+        return null;
+    });
+
+    private static final Function<Triple, IllegalArgumentException> validateNoMementoPredicates  = uncheck(x ->  {
+        if (x.getPredicate().getURI().startsWith(MEMENTO_NAMESPACE)) {
+            return new IllegalArgumentException(
```

Comment:
Is `IllegalArgumentException` the most accurate exception to be returned here? Maybe `ServerManagedPropertyException` instead?